File: test/productshotai_site_kit_test.exs
defmodule ProductshotaiSiteKitTest do
  use ExUnit.Case

  test "builds public ProductShot URLs" do
    assert ProductshotaiSiteKit.brand() == "ProductShot AI"
    assert ProductshotaiSiteKit.home_url() == "https://productshotai.app/"
    assert ProductshotaiSiteKit.workbench_url() == "https://productshotai.app/#workbench"
    assert ProductshotaiSiteKit.pricing_url() == "https://productshotai.app/#pricing"
    assert ProductshotaiSiteKit.blog_url() == "https://productshotai.app/blog/"
    assert ProductshotaiSiteKit.zh_home_url() == "https://productshotai.app/zh/"
  end

  test "follows locale URL rules" do
    assert ProductshotaiSiteKit.localized_url("en", "/blog") == "https://productshotai.app/blog/"
    assert ProductshotaiSiteKit.localized_url("zh-CN", "blog") == "https://productshotai.app/zh/blog/"
    assert_raise ArgumentError, fn -> ProductshotaiSiteKit.localized_url("fr", "/blog") end
  end

  test "exposes metadata" do
    assert ProductshotaiSiteKit.metadata().name == "ProductShot AI"
    assert ProductshotaiSiteKit.metadata().canonical_pages.home == ProductshotaiSiteKit.home_url()
  end
end

File: test/productshotai_site_kit_test.dart
import 'package:test/test.dart';
import 'package:productshotai_site_kit/productshotai_site_kit.dart';

void main() {
  test('builds public ProductShot URLs', () {
    expect(base, 'https://productshotai.app');
    expect(brand, 'ProductShot AI');
    expect(homeUrl(), 'https://productshotai.app/');
    expect(workbenchUrl(), 'https://productshotai.app/#workbench');
    expect(pricingUrl(), 'https://productshotai.app/#pricing');
    expect(blogUrl(), 'https://productshotai.app/blog/');
    expect(contactUrl(), 'https://productshotai.app/contact/');
    expect(zhHomeUrl(), 'https://productshotai.app/zh/');
  });

  test('follows locale URL rules', () {
    expect(localizedUrl('en', '/blog'), 'https://productshotai.app/blog/');
    expect(localizedUrl('zh-CN', 'blog'), 'https://productshotai.app/zh/blog/');
    expect(() => localizedUrl('fr', '/blog'), throwsArgumentError);
  });

  test('exposes metadata', () {
    final data = metadata();
    expect(data['name'], 'ProductShot AI');
    expect((data['canonicalPages'] as Map)['home'], homeUrl());
  });
}
